lib/charms/mongodb/v1/mongodb_provider.py

```diff
@@ -303,9 +303,7 @@ def _get_config(self, username: str, password: Optional[str]) -> MongoConfigurat
         if self.charm.is_role(Config.Role.MONGOS):
             mongo_args["port"] = Config.MONGOS_PORT
             if self.substrate == Config.Substrate.K8S:
-                external = self.is_external_client(relation.id)
-                mongo_args["port"] = self.charm.get_mongos_port(external)
-                mongo_args["hosts"] = self.charm.get_mongos_hosts(external)
+                mongo_args["port"] = self.charm.get_mongos_port()
```

Comment:
note - now the update of hosts + ips is done in charm side on mongos k8s 

https://github.com/canonical/mongos-k8s-operator/blob/25f2b05145c2ea4b74b92e0be0e4f7c37cc3b147/src/charm.py#L410-L436